### PR TITLE
Test, add example, and document referenced templates in a form's submit-[success|error]

### LINKED ISSUE #7118: Request for amp-form and extended template reference by id
As it is done in amp-list, allow amp-form to select an extented template like this
```

<form method="post"
      action-xhr="https://ampbyexample.com/components/amp-form/submit-form-input-text-xhr"
      target="_top"
template="amp-template-id">
    <input type="text"
        class="data-input"
        name="name"
        placeholder="Name..."
        required>
    <input type="email"
        class="data-input"
        name="email"
        placeholder="Email..."
        required>
    <input type="submit"
        value="Subscribe"
        class="button button-primary">
  </form>

<template type="amp-mustache"
        id="amp-template-id">
      {{my_variable}}
    </template>
```

### PR DESCRIPTION
**Pull request by danielrozenberg**
Fixes #7118

Turns out this feature is already implemented! Just needed documentation/test/examples

### CODE REVIEW

File: examples/forms.amp.html [hunk 1]
```diff
@@ -756,3 +756,26 @@ <h1>Here are the results for the search: {{term}}</h1>
...
+        <input name="clientId" type="hidden" value="CLIENT_ID(poll)" data-amp-replace="CLIENT_ID">
+        <input name="canonicalUrl" type="hidden" value="RANDOM and CANONICAL_URL" data-amp-replace="CANONICAL_URL RANDOM">
+        <label>
+            <span>Search for</span>
+            <input type="search" name="term" required>
+        </label>
+        <input type="submit" value="Search">
+    </fieldset>
+
+    <div submit-success template="submit_success_template"></div>
```
Comment (aghassemi) [inline examples/forms.amp.html]: Althought this works in runtime, it requires validator changes. Currently validator would complain and say `The attribute 'template' may not appear in tag 'FORM > DIV `

File: extensions/amp-form/amp-form.md [hunk 2]
```diff
@@ -234,7 +234,7 @@ The response is expected to be a valid JSON Object. For example, if the publishe
 
 Both success and error responses should have a `Content-Type: application/json` header. `submit-success` will render for all responses that has a status of `2XX`, all other statuses will render `submit-error`.
 
-Publishers can render these in a template inside their forms as follows.
+Publishers can render these in a nested template inside their forms as follows.
```
Comment (aghassemi) [inline extensions/amp-form/amp-form.md]: why `nested`?

**Comment by choumx** [comment]
Adding @honeybadgerdontcare for upcoming validator changes.

File: extensions/amp-form/amp-form.md (hunk 2 above)
Comment (danielrozenberg) [reply extensions/amp-form/amp-form.md]: :%s/nested/inlined/ better?

File: examples/forms.amp.html (hunk 1 above)
Comment (danielrozenberg) [reply examples/forms.amp.html]: @aghassemi I added this to `validator-main.protoascii`

@honeybadgerdontcare is there a way to create a validator rule to say "either `<div submit-success>` has an attribute `template` OR a child node `<template>` BUT not both"?

File: validator/validator-main.protoascii [hunk 3]
```diff
@@ -2870,6 +2870,7 @@ tags {
   spec_name: "FORM > DIV [submit-success]"
   mandatory_parent: "FORM"
   attrs: { name: "align" }
+  attrs: { name: "template" }
```
Comment (aghassemi) [inline validator/validator-main.protoascii]: Let's add new tests to `validator/testdata/feaure_tests/forms.html` and `validator/testdata/feaure_tests/forms.out` to test the validator change.

Comment (aghassemi) [inline validator/validator-main.protoascii]: sort alphabetical.

Comment (danielrozenberg) [reply validator/validator-main.protoascii]: Done

File: examples/forms.amp.html (hunk 1 above)
Comment (danielrozenberg) [reply examples/forms.amp.html]: Ping @honeybadgerdontcare

**Comment by honeybadgerdontcare** [comment]
The test files says one is invalid but it isn't being caught as invalid. Is the validator intended to catch that?

**Comment by danielrozenberg** [comment]
@honeybadgerdontcare that's what I wanted to know - if it's possible to write a validator rule that essentially says "<div submit-success>" must have exactly one of: [attribute template] OR [child node <template>]

If that's not a possible validation rule, I'll have to remove that from the test and add a caveat to the docs that say that behaviour is undefined

**Comment by honeybadgerdontcare** [comment]
@danielrozenberg Is it problematic for the non-child template case to be written as `<div submit-success-template template=...>` or maybe better `<div submit-success-template=...>`?

**Comment by danielrozenberg** [comment]
@honeybadgerdontcare I can take a look at what changes that would entail. I think it should be a relatively easy fix

**Comment by danielrozenberg** [comment]
@honeybadgerdontcare @Gregable where do we stand on the validator changes?

**Comment by honeybadgerdontcare** [comment]
Validation changes take ~2-3 weeks to get released.

File: extensions/amp-form/amp-form.md [hunk 4]
```diff
@@ -234,7 +234,7 @@ The response is expected to be a valid JSON Object. For example, if the publishe
 
 Both success and error responses should have a `Content-Type: application/json` header. `submit-success` will render for all responses that has a status of `2XX`, all other statuses will render `submit-error`.
 
-Publishers can render these in a template inside their forms as follows.
+Publishers can render these in a inlined template inside their forms as follows.
```
Comment (aghassemi) [inline extensions/amp-form/amp-form.md]: `an inlined` 

Thanks for chaining this, I prefer `inline` to reduce confusion with `nested` templates that we have been chatting about

**Comment by aghassemi** [comment]
This looks good to me but we should wait until #12670 makes it to production before merging this since it is mostly updating documentation.

File: extensions/amp-form/amp-form.md (hunk 4 above)
Comment (danielrozenberg) [reply extensions/amp-form/amp-form.md]: Done

**Comment by danielrozenberg** [comment]
Ping @aghassemi @jridgewell @honeybadgerdontcare

**Pull request merged**